### ISSUE TITLE
Prefer calling `then` and `catch` separately to combining them into a single `then`

### ISSUE THREAD

**Issue by ghost** [eslint-community/eslint-plugin-promise#217]
Essentially, `Promise.prototype.then` allows taking in two arguments to respond to both resolved and rejected promises. Essentially, `p.then(f, g)` is equivalent to `p.catch(g).then(f)`; note that the catch comes first since errors thrown by the `then` handler are *not* caught by the `catch` handler.

It would be nice to automatically replace instances like this since it makes it easier to visually scan for `catch` to check if an error is handled. It also helps emphasise the fact that the `then` is not handled by the `catch`, which isn't as apparent when it's ordered visually after the `then` handler.

**Comment by xjamundx** [comment]
Are you proposing to keep them separate AND to order them. Maybe that would be a config option. I'm not necessarily sold on the ordering thing.

**Comment by fisker** [comment]
> then(f, g) is equivalent to catch(g).then(f) always

Actually, no, they are not equivalent.

```js
Promise.resolve()
	  .then(
		  () => {throw new Error},
		  () => console.error('error caught')
	  ).catch(() => {})
```

```js
Promise.resolve()
	.then(() => {throw new Error})
	.catch(() => console.error('error caught'))
	.catch(() => {})
```

**Comment by fisker** [comment]
Sorry, I didn't read carefully, please ignore my comment above.
